.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,141 @@
+name: "Integration tests"
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+jobs:
+  integration-tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      # Need to check out repo so that 'gh pr' will know the correct repo
+      - uses: actions/checkout@v3
+
+      - name: Trigger variant analysis
+        id: trigger
+        run: |
+          if [ "$GITHUB_EVENT_NAME" == "pull_request" ]; then
+            REF="$GITHUB_HEAD_REF"
+          else
+            REF="$GITHUB_REF_NAME"
+          fi
+
+          QUERY_PACK=$(curl https://github.com/$GITHUB_REPOSITORY/releases/download/test/test_pack2.tar.gz -L | base64)
+
+          cat <<EOF >> input.json
+            {
+              "action_repo_ref": "$REF",
+              "language": "go",
+              "query_pack": "$QUERY_PACK",
+              "repositories": [
+                "docker/compose",
+                "hashicorp/terraform",
+                "github/does-not-exist"
+              ]
+            }
+          EOF
+          echo "input.json: $(cat input.json)"
+
+          RESPONSE=$(curl -H "Authorization: Bearer ${{ secrets.BOT_TOKEN }}" "https://api.github.com/repos/$GITHUB_REPOSITORY/code-scanning/codeql/variant-analyses" -X POST -d @input.json)
+          echo "Response: $RESPONSE"
+
+          ID="$(echo "$RESPONSE" | jq '.id')"
+          echo "Triggered variant analysis $ID"
+
+          if [ "$ID" == "null" ]; then
+            echo "Error triggering variant analysis"
+            exit 1
+          fi
+
+          echo "::set-output name=variant_analysis_id::$ID"
```

Comment:
Do we want to track the `set-output` deprecations as a separate issue? [The new method](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter) seems to be to write to an environment file, but we will also need to modify the `codeql-query` workflow file for that.